walk-in-interview/src/main/java/com/google/job/data/Job.java

```diff
@@ -1,18 +1,146 @@
 package com.google.job.data;
 
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
 /** Class for a job post. */
 public final class Job {
-    // TODO(issue/9): Add the feature to create a new job post
-    private String jobTitle;
+    private final String jobId;
+    // TODO(issue/25): merge the account stuff into job post.
+    private final JobStatus jobStatus;
+    private final String jobTitle;
+    private final Location jobLocation;
+    private final String jobDescription;
+    private final JobPayment jobPay;
+    private final List<String> requirements;
+    private final long postExpiry; // a timestamp
+    private final JobDuration jobDuration;
+
+    private int hashCode;
+
+    private Job(JobBuilder jobBuilder) {
+        this.jobId = jobBuilder.jobId;
+        this.jobStatus = jobBuilder.jobStatus;
+        this.jobTitle = jobBuilder.jobTitle;
+        this.jobLocation = jobBuilder.location;
+        this.jobDescription = jobBuilder.jobDescription;
+        this.jobPay = jobBuilder.jobPay;
+        this.requirements = jobBuilder.requirements;
+        this.postExpiry = jobBuilder.postExpiry;
+        this.jobDuration = jobBuilder.jobDuration;
+    }
 
     // No-argument constructor is needed to deserialize object when interacting with cloud firestore.
-    public Job() {}
+    public Job() {
+        this.jobId = "";
+        this.jobStatus = null;
```

Comment:
nulls are not valid values here (your fields are not nullable), why not set a default value instead?

Reply:
Fixed